source/Ship.cpp

```diff
@@ -3864,6 +3865,12 @@ int Ship::StepDestroyed(vector<Visual> &visuals, list<shared_ptr<Flotsam>> &flot
 			for(shared_ptr<Flotsam> &it : jettisoned)
 				it->Place(*this);
 			flotsam.splice(flotsam.end(), jettisoned);
+			for(auto &it : jettisonedFromBay)
+			{
+				it.first->Place(*this, it.second);
+				flotsam.emplace_back(it.first);
+			}
+			jettisonedFromBay.clear();
```

Comment:
Why not use splice, like above? Also this code looks quite similar to DoJettison.

Reply:
Can't use `std::list::splice` because `jettisonedFromBay` and `flotsam` do not have the same type.
`flotsam` is an `std::list<std::shared_ptr<Flotsam>>`, the same as `jettisoned`.
`jettisonedFromBay` is an `std::list<std::pair<std::shared_ptr<Flotsam>, size_t>>`.
`std::list::splice` will only work when the two containers have the same type; it wouldn't have any way to convert from `std::pair<std::shared_ptr<Flotsam>, size_t>` to `std::shared_ptr<Flotsam>`.

This code is indeed somewhat similar to `DoJettison`. Here, everything is being ejected at once, while there stuff is jettisoned one at a time. Despite the similarities, I don't think the two pieces of code can be consolidated in a way that would improve readability or maintainability.

Reply:
ah ok